Support Java 21 for Jooby 2.x

Compiling and running with jdk 21 produces this:

```
Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 65
	at io.jooby.internal.$shaded.asm.ClassReader.<init>(ClassReader.java:199)
	at io.jooby.internal.$shaded.asm.ClassReader.<init>(ClassReader.java:180)
	at io.jooby.internal.$shaded.asm.ClassReader.<init>(ClassReader.java:166)
	at io.jooby.internal.RouteAnalyzer.returnType(RouteAnalyzer.java:44)
	at io.jooby.internal.RouterImpl.start(RouterImpl.java:555)
	at io.jooby.Jooby.start(Jooby.java:865)
	at io.jooby.Server$Base.fireStart(Server.java:93)
	at io.jooby.jetty.Jetty.start(Jetty.java:96)
	at io.jooby.Jooby.start(Jooby.java:805)
	... 3 more
```

--- a/pom.xml
+++ b/pom.xml
@@ -69,7 +69,7 @@
     <jboss-modules.version>1.11.0.Final</jboss-modules.version>
 
     <!-- ASM -->
-    <asm.version>9.3</asm.version>
+    <asm.version>9.6</asm.version>
 
     <!-- Server -->
     <undertow.version>2.2.23.Final</undertow.version>

